src/main/java/org/commcare/core/network/ModernHttpRequester.java

```diff
@@ -181,14 +198,14 @@ public InputStream getResponseStream(Response<ResponseBody> response) throws IOE
         return cache.retrieveCache();
     }
 
-    /**
-     * Only gets called if response processor is supplied
-     * @return Input Stream from cache
-     * @throws IOException if an io error happens while reading or writing to cache
-     */
-    @Override
-    public InputStream getResponseStream() throws IOException {
-        return getResponseStream(response);
+    public String getApiVersion() {
+        String apiVerison = null;
+        try {
+            apiVerison = response.headers().get("x-api-current-version");
+        } catch(Exception e) {
+            //Ignore exception if API version header isn't present
```

Comment:
the call `response.headers().get("x-api-current-version")` doesn't really throw an exception even if `"x-api-current-version` is not present, as such do we need the catch here ?

Reply:
Makes sense, I removed it. I was being a little overly safe!